Add Tests

Stan-Kudri opened the pull request.

Stan-Kudri committed:
Add Test Point.

Stan-Kudri committed:
Rename method from "TrueAteFood" to "TryEatFood" and changed method "BuildBody".

Stan-Kudri committed:
Changed and Add:

-Class "Snake":
  1)Changed from "public const int ValueIncreaseSpeed" to "private readonly int _valueIncreaseSpeed";
  2)Changed constructor.

-Add test "SpeedTest".

Stan-Kudri committed:
Added a test class and a check to the "Score" class constructor.

Stan-Kudri committed:
Add and Changed:
-Added folders and moved tests to them.
-Added a test with one constructor check.
-Changed constructor from class "Food".

Stan-Kudri committed:
Add Test class "GameMapTest" and Changed clasess:

-Class "Snake": Changed method BuildBody if lenght snake == one (1).
-Class "GameMap": Changed create food in map.
-Add Extension method "Snake Creat".
-Class "Game": Calling new Extension method.

Stan-Kudri committed:
Add parameters to the method and removed the inner creation.

blowin commented on Snake/ComponentsGame/GameMap.cs:
```diff
@@ -19,17 +19,17 @@ public class GameMap
+            _food = RandomCellForFood() ?? SearchCellForFood() ?? throw new Exception("There is no empty cell for food.");
         }
 
-        public bool GameOver => _snake.ObstacleCollision() || _snake.Intersect();
```
IsGameOver

blowin commented on Snake/ComponentsGame/ItemGameMap/Snake.cs:
```diff
@@ -24,7 +24,7 @@ public Snake(int x, int y, Border border, int length)
             _widthField = border.Width;
             _heightField = border.Height;
-            (_head, _oldTail) = BuildBody(x, y);
+            (_body, _head, _oldTail) = BuildBody(x, y);
```
выше удали инициализацию, она лишняя

blowin commented on Snake/ComponentsGame/ItemGameMap/Snake.cs:
```diff
@@ -127,16 +127,25 @@ private int ClampInverted(int position, int min, int max)
             }
 
-            return (_body.Last(), _body.First());
+            return (body, body.Last(), body.First());
```
раз ты возвращаешь body, то там можешь и взять первый и последний элемент

blowin replied:
метод будет проще

blowin commented on Snake/ComponentsGame/Speed.cs:
```diff
@@ -2,27 +2,42 @@
             {
-                _numberInterval = score / _speedUpInterval;//Point interval number.
-                Value = _startSpeed - _numberInterval * ValueIncreaseSpeed;
+                _numberInterval = score / _thresholdPoints;//Point interval number.
```
перенеси комментарий над строкой

blowin commented on TestSnake/TestComponentsGame/GameMapTest.cs:
```diff
@@ -0,0 +1,54 @@
+{
+    public class GameMapTest
+    {
+        public static IEnumerable<object[]> ClassInstances()
```
название метода

blowin commented on TestSnake/TestComponentsGame/GameMapTest.cs:
```diff
@@ -0,0 +1,54 @@
+            };
+        }
+
+        public static IEnumerable<object[]> ClassNotEmptyPosition()
```
название

blowin commented on TestSnake/TestComponentsGame/GameMapTest.cs:
```diff
@@ -0,0 +1,54 @@
+        public void Move_Item_Game_Map(GameMap gameMap)
+        {
+            //Act
+            for (var i = 0; i > 9; i++)
```
лучше было создать сразу в методе, там 1 раз данные передаются, хотя бы понятнее будет

blowin commented on TestSnake/TestComponentsGame/GameMapTest.cs:
```diff
@@ -0,0 +1,54 @@
+        public void No_Food_Cell(Border border, Snake snake)
+        {
+            //Assert
+            Assert.Throws<Exception>(() => { new GameMap(border, snake); });
```
аналогично с созданием. лучше проверить что ещё в ошибке будет

blowin commented on TestSnake/TestComponentsGame/ScoreTest.cs:
```diff
@@ -0,0 +1,42 @@
+{
+    public class ScoreTest
+    {
+        public static IEnumerable<object[]> ItemScore()
```
название + нужен ли он? пока 1 раз параметры передаются

blowin commented on TestSnake/TestComponentsGame/ScoreTest.cs:
```diff
@@ -0,0 +1,42 @@
+        [MemberData(nameof(ItemScore))]
+        public void Add_Points_To_Score(Score score, int points, int expectScore)
+        {
+            //Actual
```
actual?

Stan-Kudri committed:
Changed:

-Class "Snake": Changed the constructor and output parameters of the "BuildBody" method.

-Class "GameOver": Rename method from "GameOver" to "IsGameOver".

-Class Speed: Moved above comment in method "Increase".

-Changed tests by comments:
  1)GameMapTest;
  2)ScoreTest;
  3)PointTest.

Stan-Kudri committed:
Corrected the mistake Lint.

Stan-Kudri committed:
Add test "SnakeTest" and Changed class "Snake":

-Added a test class "SnakeTest" for the "Snake" class.
-Added checks to the "Snake" class constructor.
-Added property "Head".

Stan-Kudri committed:
Add check to the "Game" class constructor.

blowin commented on Snake/ComponentsGame/ItemGameMap/Snake.cs:
```diff
@@ -17,17 +17,30 @@ public class Snake
-            (_body, _head, _oldTail) = BuildBody(x, y);
+            Direction = directions;
+            _body = BuildBody(x, y);
+            _head = _body.First();
```
честно говоря не понимаю целесообразность этих полей. почему нельзя сделать
```csharp
Point Head => _body.First();
```

blowin commented on TestSnake/PointTest.cs:
```diff
@@ -4,17 +4,20 @@ namespace TestSnake
     public class PointTest
     {
-        public static IEnumerable<object[]> TwoPoints() => new List<object[]>
+        public static IEnumerable<object[]> TwoPoints()
```
зачем метод, когда ты можешь в тесте напрямую всё создать. либо добавляй вариантов, которые проверить нужно.

blowin commented on TestSnake/TestComponentsGame/TestItemGameMap/SnakeTest.cs:
```diff
@@ -0,0 +1,96 @@
+            var snake = new Snake(startPositionX, startPositionY, border, length);
+
+            //Act
+            var expectHeadPosition = new Point(snake.Head.X + countMoveSnakeByX, snake.Head.Y);
```
ну это супер не явно. в идеале тесты НЕ ДОЛЖНЫ содержать логику. т.е ты передаёшь конкретные значения в метод (или определяешь вначале, а не высчитываешь их

Stan-Kudri committed:
Simplified tests of classes "PointTest" and "SnakeTest".

Stan-Kudri committed:
Rename folder from "Snake" -> "GameSnake".

Stan-Kudri committed:
Remove file GameSnake.sln.bak

Stan-Kudri committed:
Create .editorconfig.

Stan-Kudri committed:
Create builder.yml

Stan-Kudri committed:
GameSnake, Project - Added the ability to build a project for any OS.

Stan-Kudri committed:
Update builder.yml

Stan-Kudri committed:
Delete code-review.yml

Stan-Kudri committed:
Solution, Linter - fixed issues.

Stan-Kudri committed:
TestSnake - Added two test classes.

blowin commented on GameSnake/Extension/SnakeFieldExtension.cs:
```diff
@@ -0,0 +1,10 @@
+{
+    public static class SnakeFieldExtension
+    {
+        public static Snake Create(this int length, Border border) => new Snake((border.Width / 2) - length, border.Height / 2, border, length);
```
это не очевидно, что на int есть метод расширение, лучше передавай border в конструктор Snake и он пусть считает всё

Stan-Kudri committed:
GameSnake, SnakeFieldExtension - Changed type parameter "this".
